File: results/dnd-rest-benchmark/lifecycle-runs/20260731T062307Z_claude_claude-sonnet-5_typescript-node/src/routes/phb.ts
// Player's Handbook lookups and stateless resource math: spell slots, long
// rests, and encumbrance. No persistence.
import type { ServerResponse } from "node:http";
import { sendJson } from "../http.js";
import { isPlainObject, isValidInt } from "../validation.js";

// Only wizard level 5 is modeled; any other class/level combination is
// reported as unsupported rather than approximated.
const WIZARD_LEVEL_5_SLOTS: Record<string, number> = { "1": 4, "2": 3, "3": 2 };

export function handleSpellSlots(res: ServerResponse, body: unknown): void {
  if (!isPlainObject(body) || typeof body.class !== "string" || !isValidInt(body.level, 1, 20)) {
    sendJson(res, 400, { error: "invalid request" });
    return;
  }

  if (body.class !== "wizard" || body.level !== 5) {
    sendJson(res, 400, { error: "unsupported class or level" });
    return;
  }

  sendJson(res, 200, { class: body.class, level: body.level, slots: WIZARD_LEVEL_5_SLOTS });
}

export function handleLongRest(res: ServerResponse, body: unknown): void {
  if (
    !isPlainObject(body) ||
    !isValidInt(body.level, 1, 20) ||
    !isValidInt(body.hp_current, 0, Number.MAX_SAFE_INTEGER) ||
    !isValidInt(body.hp_max, 0, Number.MAX_SAFE_INTEGER) ||
    !isValidInt(body.hit_dice_spent, 0, Number.MAX_SAFE_INTEGER) ||
    !isValidInt(body.exhaustion_level, 0, Number.MAX_SAFE_INTEGER)
  ) {
    sendJson(res, 400, { error: "invalid request" });
    return;
  }

  const maxRecoverable = Math.max(1, Math.floor(body.level / 2));
  const hitDiceSpent = Math.max(0, body.hit_dice_spent - maxRecoverable);
  const exhaustionLevel = Math.max(0, body.exhaustion_level - 1);

  sendJson(res, 200, {
    hp_current: body.hp_max,
    hit_dice_spent: hitDiceSpent,
    exhaustion_level: exhaustionLevel,
  });
}

export function handleEquipmentLoad(res: ServerResponse, body: unknown): void {
  if (
    !isPlainObject(body) ||
    !isValidInt(body.strength, 1, Number.MAX_SAFE_INTEGER) ||
    !isValidInt(body.weight, 0, Number.MAX_SAFE_INTEGER)
  ) {
    sendJson(res, 400, { error: "invalid request" });
    return;
  }

  const capacity = body.strength * 15;

  sendJson(res, 200, { capacity, weight: body.weight, encumbered: body.weight > capacity });
}

File: results/dnd-rest-benchmark/lifecycle-runs/20260730T082348Z_claude_claude-sonnet-5_go-stdlib/phb.go
package main

import (
	"encoding/json"
	"net/http"
)

var wizardSlotsByLevel = map[int]map[string]int{
	5: {"1": 4, "2": 3, "3": 2},
}

func handleSpellSlots(w http.ResponseWriter, r *http.Request) {
	var req struct {
		Class string `json:"class"`
		Level int    `json:"level"`
	}
	if err := json.NewDecoder(r.Body).Decode(&req); err != nil {
		writeError(w, http.StatusBadRequest, "invalid request body")
		return
	}
	if req.Class != "wizard" {
		writeError(w, http.StatusBadRequest, "unsupported class")
		return
	}
	slots, ok := wizardSlotsByLevel[req.Level]
	if !ok {
		writeError(w, http.StatusBadRequest, "unsupported level")
		return
	}
	writeJSON(w, http.StatusOK, map[string]interface{}{
		"class": req.Class,
		"level": req.Level,
		"slots": slots,
	})
}

func handleLongRest(w http.ResponseWriter, r *http.Request) {
	var req struct {
		Level           int `json:"level"`
		HPCurrent       int `json:"hp_current"`
		HPMax           int `json:"hp_max"`
		HitDiceSpent    int `json:"hit_dice_spent"`
		ExhaustionLevel int `json:"exhaustion_level"`
	}
	if err := json.NewDecoder(r.Body).Decode(&req); err != nil {
		writeError(w, http.StatusBadRequest, "invalid request body")
		return
	}
	if req.Level <= 0 || req.HPMax < 0 || req.HPCurrent < 0 || req.HitDiceSpent < 0 || req.ExhaustionLevel < 0 {
		writeError(w, http.StatusBadRequest, "invalid request body")
		return
	}

	maxRecoverable := req.Level / 2
	if maxRecoverable < 1 {
		maxRecoverable = 1
	}

	hitDiceSpent := req.HitDiceSpent - maxRecoverable
	if hitDiceSpent < 0 {
		hitDiceSpent = 0
	}

	exhaustionLevel := req.ExhaustionLevel - 1
	if exhaustionLevel < 0 {
		exhaustionLevel = 0
	}

	writeJSON(w, http.StatusOK, map[string]interface{}{
		"hp_current":       req.HPMax,
		"hit_dice_spent":   hitDiceSpent,
		"exhaustion_level": exhaustionLevel,
	})
}

func handleEquipmentLoad(w http.ResponseWriter, r *http.Request) {
	var req struct {
		Strength int `json:"strength"`
		Weight   int `json:"weight"`
	}
	if err := json.NewDecoder(r.Body).Decode(&req); err != nil {
		writeError(w, http.StatusBadRequest, "invalid request body")
		return
	}
	if req.Strength < 0 || req.Weight < 0 {
		writeError(w, http.StatusBadRequest, "invalid request body")
		return
	}

	capacity := req.Strength * 15

	writeJSON(w, http.StatusOK, map[string]interface{}{
		"capacity":   capacity,
		"weight":     req.Weight,
		"encumbered": req.Weight > capacity,
	})
}
